pkg/odo/cli/component/delete.go

```diff
@@ -219,6 +236,7 @@ func NewCmdDelete(name, fullName string) *cobra.Command {
 	componentDeleteCmd.Flags().BoolVarP(&do.forceFlag, "force", "f", false, "Delete component without prompting")
 	componentDeleteCmd.Flags().BoolVarP(&do.allFlag, "all", "a", false, "Delete component and local config")
 	componentDeleteCmd.Flags().BoolVarP(&do.waitFlag, "wait", "w", false, "Wait for complete deletion of component and its dependent")
+	componentDeleteCmd.Flags().BoolVarP(&do.undeployFlag, "deploy", "d", false, "Undeploy the Devfile Kubernetes Component deployed via `odo deploy`")
```

Comment:
Could you keep the same name for the variable as the flag?
```suggestion
	componentDeleteCmd.Flags().BoolVarP(&do.deployFlag, "deploy", "d", false, "Undeploy the Devfile Kubernetes Component deployed via `odo deploy`")
```

Reply:
It can be confusing to use `deploy` in the code when wanting to undeploy, but it has to be seen as `delete deployed resources`.